### PR TITLE
Support STAC 1.0 - part 0

### PR DESCRIPTION
**Pull request by TomAugspurger**
This is a precursor PR to supporting STAC 1.0. It just updates the tests to use files from the PyStac repo, rather than sat-stac.

A followup PR will update the URLs to point to the STAC 1.0.0 versions of those files while updating the code to use PyStac 1.0.

The first two commits are just https://github.com/intake/intake-stac/pull/94 and can be ignored. https://github.com/intake/intake-stac/commit/fbf84a2f595b2e51c8c375ed2fc1746df2e6b1e6 has the actual changes.

### CODE REVIEW

File: intake_stac/catalog.py [hunk 1]
```diff
@@ -430,7 +430,9 @@ def _get_driver(self, asset):
 
         if entry_type in ['', 'null', None]:
 
-            suffix = os.path.splitext(asset.media_type)[-1]
+            suffix = '.tif'
```
Comment (TomAugspurger) [inline intake_stac/catalog.py:433]: I think the previous code was buggy. The media type is optional (https://github.com/radiantearth/stac-spec/blob/master/item-spec/item-spec.md#asset-object), so we shouldn't assume it's a string.

File: intake_stac/tests/test_catalog.py [hunk 2]
```diff
@@ -35,7 +39,12 @@ def pystac_cat():
 
 @pytest.fixture(scope='module')
 def pystac_col():
-    return pystac.Collection.from_file(col_url)
+    col = pystac.Collection.from_file(col_url)
+    # fix some linkrot
```
Comment (TomAugspurger) [inline intake_stac/tests/test_catalog.py:43]: One of the links in this file is unfortunately valid, so we hit an error when PyStac tried to load all the children. We're pointing to the tag on GitHub, so we can't change the source. Probably OK to just fix it here.

File: intake_stac/tests/test_catalog.py [hunk 3]
```diff
@@ -53,180 +62,199 @@ def intake_stac_cat():
...
-    # NOTE: note sure why asset.metadata has 'catalog_dir' key ?
-    # assert d['metadata'] == asset.metadata
-
-
-def test_asset_missing_type(pystac_item):
-    key = 'B1'
-    asset = pystac_item.assets.get('B1')
-    asset.media_type = ''
-    with pytest.warns(Warning, match='STAC Asset'):
+class TestCatalog:
```
Comment (TomAugspurger) [inline intake_stac/tests/test_catalog.py:65]: Just nesting these for ease of testing a subset. I can change it back to top-level if you want. I'll call out notable changes below.

File: intake_stac/tests/test_catalog.py [hunk 4]
```diff
@@ -53,180 +62,199 @@ def intake_stac_cat():
...
+        assert intake_stac_cat.name == 'test'
+
+
+class TestCollection:
+    def test_cat_from_collection(self, pystac_col):
+        cat = StacCollection(pystac_col)
+        subcat_name = 'landsat-8-l1'
+        assert cat.name == pystac_col.id
+        assert subcat_name in cat
+        # This is taking way too long
```
Comment (TomAugspurger) [inline intake_stac/tests/test_catalog.py:123]: I've disabled the rest of it test. It was taking a while. Is that expected?

File: intake_stac/tests/test_catalog.py [hunk 5]
```diff
@@ -13,15 +13,19 @@
...
+    'data-files/catalogs/test-case-1/catalog.json'
+)
+col_url = (
+    'https://raw.githubusercontent.com/stac-utils/pystac/v1.0.0-beta.2/tests/data-files/'
+    'examples/1.0.0-beta.2/collection-spec/examples/landsat-collection.json'
+)
+item_url = (
+    'https://raw.githubusercontent.com/stac-utils/pystac/v0.5.6/tests/data-files/'
+    'eo/eo-landsat-example.json'
+)
```
Comment (scottyhq) [inline intake_stac/tests/test_catalog.py:28]: i think we should switch all tests to use valid STAC=1.0 examples:
```
# 1.0.0 tag doesn't exist yet, use main for testing
https://raw.githubusercontent.com/stac-utils/pystac/main/tests/data-files/examples/1.0.0/catalog.json
https://raw.githubusercontent.com/stac-utils/pystac/main/tests/data-files/eo/eo-collection.json
```

File: intake_stac/tests/test_catalog.py (hunk 2 above)
Comment (scottyhq) [reply intake_stac/tests/test_catalog.py:43]: Yeah, I've noticed a lot of existing test catalogs don't worry so much about the validity of asset links, whereas with intake-stac we do want to test against real assets. maybe we embed a couple test catalogs in this repository or update and use https://github.com/stac-utils/stac-examples/issues/1 ?

Comment (scottyhq) [reply intake_stac/tests/test_catalog.py:43]: Note that `https://landsat-stac.s3.amazonaws.com/landsat-8-l1/catalog.json` will be deprecated on July 1 (https://registry.opendata.aws/landsat-8/)

File: intake_stac/tests/test_catalog.py (hunk 4 above)
Comment (scottyhq) [reply intake_stac/tests/test_catalog.py:123]: has the catalog changed? does it hang if the item id doesn't exist? it shouldn't take long by the look of things

**Comment by scottyhq** [comment]
Thanks for starting this @TomAugspurger ! I left some comments trying to figure out whether to use external examples for testing, or if we might want to put our own example STAC=1.0 examples in this repository (with valid asset / subcatalog links)

**Comment by TomAugspurger** [comment]
I think having our own STAC catalogs, with just the data we need for testing, makes sense. Now that we're at STAC 1.0, this seems more feasible. I'll look into that.

**Comment by TomAugspurger** [comment]
OK, updated to use some local files, rather than opening STAC objects from remote URLs.

I think the `stack_bands` might be a bit under-tested, but hopefully I got it right. The items I'm using came from the Planetary Computer, so URLs to data files in blob storage have to be signed to download. I don't think we want to do that in our unit tests, but here's a manual tests:

```python
In [1]: import rasterio

In [2]: import pystac, planetary_computer, intake

In [3]: item = planetary_computer.sign_assets(pystac.read_file("https://planetarycomputer.microsoft.com/api/stac/v1/collections/sentinel-2-l2a/items/S2B_MSIL2A_20210610T115639_N0212_R066_T33
   ...: XWC_20210613T190000.SAFE"))

In [5]: intake_item = intake.open_stac_item(item)

In [6]: ds = intake_item.stack_bands(["B02", "B03"])().to_dask()

In [7]: ds
Out[7]:
<xarray.DataArray (band: 2, y: 10980, x: 10980)>
dask.array<concatenate, shape=(2, 10980, 10980), dtype=uint16, chunksize=(1, 10980, 10980), chunktype=numpy.ndarray>
Coordinates:
  * band     (band) int64 1 1
  * y        (y) float64 8.3e+06 8.3e+06 8.3e+06 ... 8.19e+06 8.19e+06 8.19e+06
  * x        (x) float64 5e+05 5e+05 5e+05 ... 6.098e+05 6.098e+05 6.098e+05
Attributes:
    transform:      (10.0, 0.0, 499980.0, 0.0, -10.0, 8300040.0)
    crs:            +init=epsg:32633
    res:            (10.0, 10.0)
    is_tiled:       1
    nodatavals:     (nan,)
    scales:         (1.0,)
    offsets:        (0.0,)
    AREA_OR_POINT:  Area
```

**Review by scottyhq — APPROVED** [review]
Looks great @TomAugspurger ! Would be happy to merge this as is. We can add some additional tests in future PRs and hopefully release 1.0.0 shortly after pystac 1.0.0 ?

**Comment by TomAugspurger** [comment]
Yeah, releasing shortly after pystac 1.0 is my hope.

**Pull request merged**